vortex-serde/src/layouts/read/stream.rs

```diff
@@ -74,107 +74,127 @@ impl<R: VortexReadAt> LayoutBatchStream<R> {
 
 type StreamStateFuture<R> = BoxFuture<'static, VortexResult<(R, Vec<(MessageId, Bytes)>)>>;
 
-#[derive(Default)]
+/// State of vortex file stream
+///
+/// The flow starts with `AddSplits` which produces all horizontal row splits in the file
+/// Main read loop goes from `NextSplit` -> `Filter` (if there's filter) -> `Read`
+/// `Filter` and `Read` states transition to `Reading` when they're blocked on an io operation which resumes back to
+/// the previous state.
 enum StreamingState<R> {
-    #[default]
-    Init,
-    FilterInit,
-    Reading(StreamStateFuture<R>),
-    FilterReading(StreamStateFuture<R>),
-    Decoding(Array),
+    AddSplits,
+    NextSplit,
+    Filter,
+    Read,
+    Reading(StreamStateFuture<R>, bool),
```

Comment:
Can this be `Reading { future: StreamStateFuture<R>, XXX: bool }` - it's not clear what bool is right now